contracts/smart-account/interfaces/modules/SessionValidationModules/IABISessionValidationModule.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import "../ISessionValidationModule.sol";
+
+abstract contract IABISessionValidationModule is ISessionValidationModule {
+    enum Condition {
+        EQUAL,
+        LESS_THAN_OR_EQUAL,
+        LESS_THAN,
+        GREATER_THAN_OR_EQUAL,
+        GREATER_THAN,
+        NOT_EQUAL
+    }
+
+    struct Rule {
+        uint256 offset;
+        bytes32 value;
+        Condition condition;
+    }
+
+    struct Permission {
+        address destinationContract;
+        bytes4 selector;
+        uint256 valueLimit;
+        Rule[] rules;
+    }
```

Comment:
when this gets ABI-encoded, each field gets promoted to a 32-byte representation.

For fields like `bytes4 selector`, `Condition condition` and to a degree `address destinationContract` and `uint256 valueLimit`, these 32-byte representations will be "mostly" empty adding to the calldata of the signature.

I would recommend encoding them as closely as possible to save on calldata costs.

Reply:
great idea. switched to bytes array instead of structs

Reply:
save 500-1000 compute gas as well as calldata cost reduction